### ISSUE TITLE
Autocomplete: accessibility issues

### ISSUE THREAD

**Issue by dryton** [formkit/formkit#923]
### Reproduction

https://formkit.com/inputs/autocomplete

### Describe the bug

A couple of issues are being flagged up by axe automated accessibility tests in relation to the listbox button used in the autocomplete component:

1. The autocomplete listbox button is missing an accessible name, e.g. via text or an `aria-label`/`aria-labelledby` attribute 
> 'aria-command-name' │ 'serious' │ 'Ensures every ARIA button, link and menuitem has an accessible name' 

Reference: https://dequeuniversity.com/rules/axe/html/3.5/aria-command-name

2. Listbox button has a positive tab index
>  'tabindex'  │ 'serious' │ 'Ensures tabindex attribute values are not greater than 0'  

Reference: https://dequeuniversity.com/rules/axe/html/3.5/tabindex

### Environment

• OS: any
• Browser: all

**Comment by sashamilenkovic** [comment]
@dryton Greatly appreciate the feedback! I've attempted to address this on the next tag of FormKit Pro [0.115.2-fc79935](https://www.npmjs.com/package/@formkit/pro/v/0.115.2-fc79935).

**Comment by davidsandoz** [comment]
@sashamilenkovic I'm still getting an error from the axe DevTools because of a missing accessible name.

I'm running FormKit Pro v0.119.7. Am I missing something, or does this still need to be implemented?

<img width="797" alt="Screenshot 2024-02-05 at 11 11 20" src="https://github.com/formkit/formkit/assets/1535702/b1b693f5-c676-46d9-84fb-6cdb89f12794">

**Comment by sashamilenkovic** [comment]
@davidsandoz Hmmmmm, not sure if there was a regression or what. We will take a look at this.

**Comment by sashamilenkovic** [comment]
@davidsandoz We've addressed this by adding `aria-label` attribute to the listboxSection at FormKit Pro v0.121.6. I'l leave this open for now and wait for your feedback.

**Comment by davidsandoz** [comment]
@sashamilenkovic Yes, that's better, thank you!

This brought up other potential issues:
- The label value stays to “Open” when the listbox is open, while I suppose it should change to “Close”.
- Using the keyboard, I can focus on the listbox button, but I cannot open it with the keyboard, neither with the space key nor the enter key.

Please let me know if I should rather open new issues about those.

**Comment by sashamilenkovic** [comment]
@davidsandoz That would be great if you wouldn't mind opening a new issue for those.

**Comment by davidsandoz** [comment]
@sashamilenkovic I've created #1201 for the second issue.

For the first one (`aria-label` not changing to _Close_), isn't it introduced with the fix brought for this issue? And thus, wouldn't it make sense to have it fixed here? If not, I'll create another issue specifically for this, as it is different from #1201.

**Comment by sashamilenkovic** [comment]
@davidsandoz you're probably right, let's keep this one open.

**Comment by sashamilenkovic** [comment]
@davidsandoz At FormKit Pro v0.121.8, the `aria-label` should now alternate between Open and Close depending on whether the listbox is open.
